src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -222,13 +222,9 @@ private void initSearchModifierButtons() {
         });
 
         fulltextButton.setSelected(searchPreferences.isFulltext());
-        searchPreferences.getObservableSearchFlags().addListener((SetChangeListener<SearchRules.SearchFlags>) c -> {
-            fulltextButton.setManaged(searchPreferences.isFulltext());
-        });
-        fulltextButton.setManaged(searchPreferences.isFulltext());
-
         fulltextButton.setTooltip(new Tooltip(Localization.lang("Fulltext search")));
         initSearchModifierButton(fulltextButton);
+        EasyBind.subscribe(preferencesService.getFilePreferences().fulltextIndexLinkedFilesProperty(), value -> fulltextButton.setDisable(!value));
```

Comment:
This is a nice idea! Additionally, it should set the button state to "false" if the fulltext search is deactivated (and only then)